server/src/main/java/org/candlepin/auth/ConsumerAuth.java

```diff
@@ -51,30 +49,35 @@
         this.i18nProvider = i18nProvider;
     }
 
-    public ConsumerPrincipal createPrincipal(String consumerUuid) {
-        ConsumerPrincipal principal = null;
-
-        if (consumerUuid != null) {
-            // If this UUID has been deleted, return a 410.
-            if (deletedConsumerCurator.countByConsumerUuid(consumerUuid) > 0) {
-                log.debug("Key {} is deleted, throwing GoneException", consumerUuid);
-                throw new GoneException(
-                    i18nProvider.get().tr("Unit {0} has been deleted", consumerUuid), consumerUuid);
-            }
-
-            Consumer consumer = this.consumerCurator.getConsumer(consumerUuid);
-
-            if (consumer != null) {
-                Owner owner = this.ownerCurator.findOwnerById(consumer.getOwnerId());
-                principal = new ConsumerPrincipal(consumer, owner);
+    /**
+     * Creates a principal according to the {@link Consumer} with the given
+     * consumerUuid.
+     *
+     * @param consumerUuid requested consumer
+     * @return created principal
+     */
+    public ConsumerPrincipal createPrincipal(final String consumerUuid) {
+        if (consumerUuid == null) {
+            return null;
+        }
 
-                if (log.isDebugEnabled() && principal != null) {
-                    log.debug("principal created for consumer {}", principal.getConsumer().getUuid());
-                }
+        final Consumer consumer = this.consumerCurator.getConsumer(consumerUuid);
+        if (consumer == null) {
+            if (wasDeleted(consumerUuid)) {
+                throw new GoneException(i18nProvider.get()
+                    .tr("Unit {0} has been deleted", consumerUuid), consumerUuid);
             }
+            return null;
         }
 
+        final Owner owner = this.ownerCurator.findOwnerById(consumer.getOwnerId());
+        final ConsumerPrincipal principal = new ConsumerPrincipal(consumer, owner);
+        log.debug("principal created for consumer {}", principal.getConsumer().getUuid());
         return principal;
     }
 
+    private boolean wasDeleted(final String consumerUuid) {
```

Comment:
This is stylistic thing, but I'm not seeing the value of pulling small one-off operations like this out of the main code. Depending on how the compiler handles this, we may inherit the overhead of a method call to get some self-documenting code, when a comment can achieve the same thing without the risk of any additional code or runtime complexity.

Similarly, declaring variables as final doesn't have a ton of value in this context outside of programmer hints (especially in Java8 where the inner class ref doesn't require it anymore).